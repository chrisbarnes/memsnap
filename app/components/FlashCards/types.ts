export interface MathQuestion {
  x: number;
  y: number;
  operand: string;
}

export interface MathFlashCard {
  id: number;
  sideA: MathQuestion;
  sideB: number;
}

export interface IMathFlashCardProps extends MathFlashCard {
  displayAnswer: boolean;
  isCorrect: boolean;
  wasSeen: boolean;
}

export interface IFlashCardsProps {
  cards: MathFlashCard[];
}
