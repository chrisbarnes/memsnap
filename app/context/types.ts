export interface CardSet {
  id: number;
  name: string;
  description: string;
  cards: any[];
  type: string;
}

export interface ICardSetContext {
  cardSet: CardSet | null;
  setCardSet: (cardSet: CardSet) => void;
}

export interface ICardSetProviderProps {
  children: React.ReactNode;
}
