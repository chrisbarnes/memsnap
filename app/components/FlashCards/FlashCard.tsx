import type { IMathFlashCardProps } from "./types";

export const FlashCard = ({
  displayAnswer,
  sideA,
  sideB,
}: IMathFlashCardProps) => {
  return (
    <div
      className={`group ${
        displayAnswer ? "flip" : ""
      } relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]`}
    >
      <section
        className={`group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700`}
      >
        <div className="leading-[112px]">{sideA.x}</div>
        <span className="absolute left-12 top-52 text-6xl pb-4">
          {sideA.operand}
        </span>
        <div className="leading-[112px] pb-4 border-b-8 border-black dark:border-white">
          {sideA.y}
        </div>
      </section>
      <section
        className={`group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700`}
      >
        {sideB}
      </section>
    </div>
  );
};
