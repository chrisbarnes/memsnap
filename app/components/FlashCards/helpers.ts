import type { MathFlashCard } from "./types";

// function that generates flash cards based on the numbers array passed in
// it should generate all possible combinations of the numbers array passed in
// multiplied by numbers 0 - 12
export const generateFlashCards = (numbers: number[]): MathFlashCard[] => {
  const flashCards: MathFlashCard[] = [];

  for (let i = 0; i <= 12; i++) {
    for (let j = 0; j < numbers.length; j++) {
      flashCards.push({
        id: flashCards.length + 1,
        sideA: {
          x: numbers[j],
          y: i,
          operand: "x",
        },
        sideB: numbers[j] * i,
      });
    }
  }

  return flashCards;
};

// export const generateFlashCards = (numbers: number[]): MathFlashCard[] => {
//   const flashCards: MathFlashCard[] = [];

//   for (let i = 0; i < 12; i++) {
//     for (let j = 0; j < numbers.length; j++) {
//       flashCards.push({
//         id: flashCards.length + 1,
//         sideA: {
//           x: numbers[i],
//           y: numbers[j],
//           operand: "x",
//         },
//         sideB: numbers[i] * numbers[j],
//       });
//     }
//   }

//   return flashCards;
// };

// export const generateFlashCards = (): MathFlashCard[] => {
//   const flashCards: MathFlashCard[] = [];

//   for (let i = 0; i <= 12; i++) {
//     for (let j = 0; j <= 12; j++) {
//       flashCards.push({
//         id: flashCards.length + 1,
//         sideA: {
//           x: i,
//           y: j,
//           operand: "x",
//         },
//         sideB: i * j,
//       });
//     }
//   }

//   return flashCards;
// };

// randomize the order of the flash cards
export const shuffleFlashCards = (
  flashCards: MathFlashCard[]
): MathFlashCard[] => {
  return flashCards.sort(() => Math.random() - 0.5);
};
