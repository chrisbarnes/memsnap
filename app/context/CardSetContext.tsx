import { createContext, useContext, useState } from "react";
import type { CardSet, ICardSetContext, ICardSetProviderProps } from "./types";

export const CardSetContext = createContext<ICardSetContext>({
  cardSet: null,
  setCardSet: () => {},
});

export const CardSetProvider = ({ children }: ICardSetProviderProps) => {
  const [cardSet, setCardSet] = useState<CardSet | null>(null);

  return (
    <CardSetContext.Provider value={{ cardSet, setCardSet }}>
      {children}
    </CardSetContext.Provider>
  );
};

export const useCardSet = () => {
  const context = useContext(CardSetContext);

  if (context === undefined) {
    throw new Error("useCardSet must be used within a CardSetProvider");
  }

  return context;
};
