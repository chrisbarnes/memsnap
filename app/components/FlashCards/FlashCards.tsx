import React, { useEffect, useState } from "react";
import { FlashCard } from "./FlashCard";
import { useCardSet } from "~/context/CardSetContext";
import type { IMathFlashCardProps } from "./types";
import { FlashCardControls } from "../FlashCardControls/FlashCardControls";

const CARD_ADVANCE_DELAY = 150;

export const FlashCards = () => {
  const { cardSet } = useCardSet();
  const { cards } = cardSet || { cards: [] };
  const [cardIndex, setCardIndex] = useState<number>(0);
  const [isDone, setIsDone] = useState<boolean>(false);
  const [displayAnswer, setDisplayAnswer] = useState<boolean>(false);
  const [renderedCards] = useState<IMathFlashCardProps[]>(
    cards.map((card) => ({
      ...card,
      displayAnswer: false,
      isCorrect: false,
      wasSeen: false,
    }))
  );
  const next = () => {
    if (cardIndex === cards.length - 1) {
      setIsDone(true);
    }

    setDisplayAnswer(false);

    if (displayAnswer) {
      setTimeout(() => {
        setCardIndex(cardIndex + 1);
      }, CARD_ADVANCE_DELAY);

      return;
    }

    setCardIndex(cardIndex + 1);
  };
  const previous = () => {
    if (cardIndex > 0) {
      setDisplayAnswer(false);

      if (displayAnswer) {
        setTimeout(() => {
          setCardIndex(cardIndex - 1);
        }, CARD_ADVANCE_DELAY);

        return;
      }

      setCardIndex(cardIndex - 1);
    }
  };

  useEffect(() => {
    const handleKeyDown = (event: KeyboardEvent) => {
      if (!isDone) {
        if (event.code === "ArrowRight") {
          next();
        }

        if (event.code === "ArrowLeft" && cardIndex > 0) {
          previous();
        }

        if (event.code === "ArrowUp" || event.code === "ArrowDown") {
          setDisplayAnswer(!displayAnswer);
        }
      }
    };

    window.addEventListener("keydown", handleKeyDown);

    return () => {
      window.removeEventListener("keydown", handleKeyDown);
    };
  });

  return (
    <React.Fragment>
      {!isDone && (
        <div className="flex flex-col">
          <p
            className="mx-4 text-sm text-slate-500 text-right mb-2"
            data-testid="card-set-status"
          >
            {cardIndex + 1}/{renderedCards.length}
          </p>
          {renderedCards[cardIndex] && (
            <FlashCard
              {...renderedCards[cardIndex]}
              displayAnswer={displayAnswer}
            />
          )}
          <FlashCardControls
            onPrevious={previous}
            onNext={next}
            onShowAnswer={() => setDisplayAnswer(!displayAnswer)}
          />
          <p className="mx-4 mb-2 text-sm dark:text-white">
            Use the left and right arrow keys to go forward and back through the
            cards. Press up or down to flip the card.
          </p>
        </div>
      )}
      {isDone && (
        <p
          className="text-5xl text-gray-900 dark:text-white"
          data-testid="done"
        >
          Done!
        </p>
      )}
    </React.Fragment>
  );
};
