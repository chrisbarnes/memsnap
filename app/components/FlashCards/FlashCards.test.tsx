import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { FlashCards } from "./FlashCards";
import { CardSetContext } from "~/context/CardSetContext";
import type { CardSet } from "~/context/types";

describe("FlashCards", () => {
  const mockCardSet: CardSet = {
    cards: [
      {
        sideA: {
          x: 1,
          y: 2,
          operand: "+",
        },
        sideB: 3,
      },
      {
        sideA: {
          x: 2,
          y: 2,
          operand: "+",
        },
        sideB: 4,
      },
    ],
    description: "Mock Card Set",
    id: 3,
    name: "Mock Card Set",
    type: "math",
  };

  it("should render and have keyboard functionality", async () => {
    const user = userEvent.setup();
    const { container } = render(
      <CardSetContext.Provider
        value={{ cardSet: mockCardSet, setCardSet: jest.fn() }}
      >
        <FlashCards />
      </CardSetContext.Provider>
    );
    expect(container).toMatchSnapshot();

    const cardSetStatus = screen.getByTestId("card-set-status");
    expect(cardSetStatus).toHaveTextContent("1/2");

    const sideAX = screen.getByTestId("side-a-x");

    await user.keyboard("{arrowright}");

    expect(sideAX).toHaveTextContent("2");

    await user.keyboard("{arrowleft}");

    expect(sideAX).toHaveTextContent("1");

    await user.keyboard("{arrowup}");

    expect(container).toMatchSnapshot();

    await user.keyboard("{arrowdown}");

    expect(container).toMatchSnapshot();

    await user.keyboard("{arrowright}");
    await user.keyboard("{arrowright}");

    expect(sideAX).not.toBeInTheDocument();

    const done = screen.getByTestId("done");

    expect(done).toBeInTheDocument();
  });

  it("should render and have button functionality", async () => {
    const user = userEvent.setup();
    const { container } = render(
      <CardSetContext.Provider
        value={{ cardSet: mockCardSet, setCardSet: jest.fn() }}
      >
        <FlashCards />
      </CardSetContext.Provider>
    );
    expect(container).toMatchSnapshot();

    const cardSetStatus = screen.getByTestId("card-set-status");
    expect(cardSetStatus).toHaveTextContent("1/2");

    const sideAX = screen.getByTestId("side-a-x");

    const nextButton = screen.getByTestId("next-button");
    const previousButton = screen.getByTestId("previous-button");
    const showAnswerButton = screen.getByTestId("show-answer-button");

    await user.click(nextButton);

    expect(sideAX).toHaveTextContent("2");

    await user.click(previousButton);

    expect(sideAX).toHaveTextContent("1");

    await user.click(showAnswerButton);

    expect(container).toMatchSnapshot();

    await user.click(showAnswerButton);

    expect(container).toMatchSnapshot();

    await user.click(nextButton);
    await user.click(nextButton);

    expect(sideAX).not.toBeInTheDocument();

    const done = screen.getByTestId("done");

    expect(done).toBeInTheDocument();
  });
});
