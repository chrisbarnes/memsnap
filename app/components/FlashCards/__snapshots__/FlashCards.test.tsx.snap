// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashCards should render and have button functionality 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="mx-4 text-sm text-slate-500 text-right mb-2"
      data-testid="card-set-status"
    >
      1
      /
      2
    </p>
    <div
      class="group  relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]"
    >
      <section
        class="group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="question"
      >
        <div
          class="leading-[112px]"
          data-testid="side-a-x"
        >
          1
        </div>
        <span
          class="absolute left-12 top-52 text-6xl pb-4"
        >
          +
        </span>
        <div
          class="leading-[112px] pb-4 border-b-8 border-black dark:border-white"
        >
          2
        </div>
      </section>
      <section
        class="group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="answer"
      >
        3
      </section>
    </div>
    <div
      class="inline-flex justify-center mb-2 rounded-md shadow-sm"
    >
      <button
        class="px-4 py-2 text-sm font-medium text-blue-700 bg-white border border-gray-200 rounded-s-lg hover:bg-gray-100 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="previous-button"
        type="button"
      >
        Previous
      </button>
      <button
        class="px-4 py-2 text-sm font-medium text-gray-900 bg-white border-t border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="show-answer-button"
        type="button"
      >
        Show Answer
      </button>
      <button
        class="min-w-[91px] px-4 py-2 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-e-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="next-button"
        type="button"
      >
        Next
      </button>
    </div>
    <p
      class="mx-4 mb-2 text-sm dark:text-white"
    >
      Use the left and right arrow keys to go forward and back through the cards. Press up or down to flip the card.
    </p>
  </div>
</div>
`;

exports[`FlashCards should render and have button functionality 2`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="mx-4 text-sm text-slate-500 text-right mb-2"
      data-testid="card-set-status"
    >
      1
      /
      2
    </p>
    <div
      class="group flip relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]"
    >
      <section
        class="group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="question"
      >
        <div
          class="leading-[112px]"
          data-testid="side-a-x"
        >
          1
        </div>
        <span
          class="absolute left-12 top-52 text-6xl pb-4"
        >
          +
        </span>
        <div
          class="leading-[112px] pb-4 border-b-8 border-black dark:border-white"
        >
          2
        </div>
      </section>
      <section
        class="group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="answer"
      >
        3
      </section>
    </div>
    <div
      class="inline-flex justify-center mb-2 rounded-md shadow-sm"
    >
      <button
        class="px-4 py-2 text-sm font-medium text-blue-700 bg-white border border-gray-200 rounded-s-lg hover:bg-gray-100 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="previous-button"
        type="button"
      >
        Previous
      </button>
      <button
        class="px-4 py-2 text-sm font-medium text-gray-900 bg-white border-t border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="show-answer-button"
        type="button"
      >
        Show Answer
      </button>
      <button
        class="min-w-[91px] px-4 py-2 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-e-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="next-button"
        type="button"
      >
        Next
      </button>
    </div>
    <p
      class="mx-4 mb-2 text-sm dark:text-white"
    >
      Use the left and right arrow keys to go forward and back through the cards. Press up or down to flip the card.
    </p>
  </div>
</div>
`;

exports[`FlashCards should render and have button functionality 3`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="mx-4 text-sm text-slate-500 text-right mb-2"
      data-testid="card-set-status"
    >
      1
      /
      2
    </p>
    <div
      class="group  relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]"
    >
      <section
        class="group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="question"
      >
        <div
          class="leading-[112px]"
          data-testid="side-a-x"
        >
          1
        </div>
        <span
          class="absolute left-12 top-52 text-6xl pb-4"
        >
          +
        </span>
        <div
          class="leading-[112px] pb-4 border-b-8 border-black dark:border-white"
        >
          2
        </div>
      </section>
      <section
        class="group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="answer"
      >
        3
      </section>
    </div>
    <div
      class="inline-flex justify-center mb-2 rounded-md shadow-sm"
    >
      <button
        class="px-4 py-2 text-sm font-medium text-blue-700 bg-white border border-gray-200 rounded-s-lg hover:bg-gray-100 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="previous-button"
        type="button"
      >
        Previous
      </button>
      <button
        class="px-4 py-2 text-sm font-medium text-gray-900 bg-white border-t border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="show-answer-button"
        type="button"
      >
        Show Answer
      </button>
      <button
        class="min-w-[91px] px-4 py-2 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-e-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="next-button"
        type="button"
      >
        Next
      </button>
    </div>
    <p
      class="mx-4 mb-2 text-sm dark:text-white"
    >
      Use the left and right arrow keys to go forward and back through the cards. Press up or down to flip the card.
    </p>
  </div>
</div>
`;

exports[`FlashCards should render and have keyboard functionality 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="mx-4 text-sm text-slate-500 text-right mb-2"
      data-testid="card-set-status"
    >
      1
      /
      2
    </p>
    <div
      class="group  relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]"
    >
      <section
        class="group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="question"
      >
        <div
          class="leading-[112px]"
          data-testid="side-a-x"
        >
          1
        </div>
        <span
          class="absolute left-12 top-52 text-6xl pb-4"
        >
          +
        </span>
        <div
          class="leading-[112px] pb-4 border-b-8 border-black dark:border-white"
        >
          2
        </div>
      </section>
      <section
        class="group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="answer"
      >
        3
      </section>
    </div>
    <div
      class="inline-flex justify-center mb-2 rounded-md shadow-sm"
    >
      <button
        class="px-4 py-2 text-sm font-medium text-blue-700 bg-white border border-gray-200 rounded-s-lg hover:bg-gray-100 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="previous-button"
        type="button"
      >
        Previous
      </button>
      <button
        class="px-4 py-2 text-sm font-medium text-gray-900 bg-white border-t border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="show-answer-button"
        type="button"
      >
        Show Answer
      </button>
      <button
        class="min-w-[91px] px-4 py-2 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-e-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="next-button"
        type="button"
      >
        Next
      </button>
    </div>
    <p
      class="mx-4 mb-2 text-sm dark:text-white"
    >
      Use the left and right arrow keys to go forward and back through the cards. Press up or down to flip the card.
    </p>
  </div>
</div>
`;

exports[`FlashCards should render and have keyboard functionality 2`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="mx-4 text-sm text-slate-500 text-right mb-2"
      data-testid="card-set-status"
    >
      1
      /
      2
    </p>
    <div
      class="group flip relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]"
    >
      <section
        class="group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="question"
      >
        <div
          class="leading-[112px]"
          data-testid="side-a-x"
        >
          1
        </div>
        <span
          class="absolute left-12 top-52 text-6xl pb-4"
        >
          +
        </span>
        <div
          class="leading-[112px] pb-4 border-b-8 border-black dark:border-white"
        >
          2
        </div>
      </section>
      <section
        class="group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="answer"
      >
        3
      </section>
    </div>
    <div
      class="inline-flex justify-center mb-2 rounded-md shadow-sm"
    >
      <button
        class="px-4 py-2 text-sm font-medium text-blue-700 bg-white border border-gray-200 rounded-s-lg hover:bg-gray-100 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="previous-button"
        type="button"
      >
        Previous
      </button>
      <button
        class="px-4 py-2 text-sm font-medium text-gray-900 bg-white border-t border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="show-answer-button"
        type="button"
      >
        Show Answer
      </button>
      <button
        class="min-w-[91px] px-4 py-2 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-e-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="next-button"
        type="button"
      >
        Next
      </button>
    </div>
    <p
      class="mx-4 mb-2 text-sm dark:text-white"
    >
      Use the left and right arrow keys to go forward and back through the cards. Press up or down to flip the card.
    </p>
  </div>
</div>
`;

exports[`FlashCards should render and have keyboard functionality 3`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="mx-4 text-sm text-slate-500 text-right mb-2"
      data-testid="card-set-status"
    >
      1
      /
      2
    </p>
    <div
      class="group  relative w-80 h-[440px] mb-2 text-9xl text-gray-900 dark:text-white font-extrabold overflow-hidden [perspective-1000px]"
    >
      <section
        class="group-[.flip]:[transform:rotateY(180deg)] absolute left-0 w-full p-12 text-right transition-all duration-200 [transform-style:preserve-3d] [backface-visibility:hidden] rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="question"
      >
        <div
          class="leading-[112px]"
          data-testid="side-a-x"
        >
          1
        </div>
        <span
          class="absolute left-12 top-52 text-6xl pb-4"
        >
          +
        </span>
        <div
          class="leading-[112px] pb-4 border-b-8 border-black dark:border-white"
        >
          2
        </div>
      </section>
      <section
        class="group-[.flip]:[transform:rotateY(0deg)] absolute left-0 w-full p-12 text-center flex flex-col justify-center [backface-visibility:hidden] [transform:rotateY(-180deg)] [transform-style:preserve-3d] transition-all duration-200 rounded shadow-lg border-[1px] border-gray-100 dark:border-gray-950 h-[440px] dark:bg-gray-700"
        data-testid="answer"
      >
        3
      </section>
    </div>
    <div
      class="inline-flex justify-center mb-2 rounded-md shadow-sm"
    >
      <button
        class="px-4 py-2 text-sm font-medium text-blue-700 bg-white border border-gray-200 rounded-s-lg hover:bg-gray-100 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="previous-button"
        type="button"
      >
        Previous
      </button>
      <button
        class="px-4 py-2 text-sm font-medium text-gray-900 bg-white border-t border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="show-answer-button"
        type="button"
      >
        Show Answer
      </button>
      <button
        class="min-w-[91px] px-4 py-2 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-e-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-blue-500 dark:focus:text-white"
        data-testid="next-button"
        type="button"
      >
        Next
      </button>
    </div>
    <p
      class="mx-4 mb-2 text-sm dark:text-white"
    >
      Use the left and right arrow keys to go forward and back through the cards. Press up or down to flip the card.
    </p>
  </div>
</div>
`;
